### PR TITLE
Revamped ElytraReplace

### PR DESCRIPTION
**Pull request by Nathan-Webb**
**Describe the pull**
This overhauls the current ElytraReplace code and makes it somewhat readable and usable.

**Describe how this pull is helpful**
I made the code a bit more modular, made some boolean logic easier to follow, and maybe fixed a bug? 

**Additional context**
I am a bit unfamiliar with kotlin, so if there are any convention / best practices violations please don't hesitate to let me know. Also if there is a style guide that I missed I'll be happy to modify my code accordingly. 

I have already tested it out in a singleplayer world, and it should work in a multiplayer world as I borrowed the swapping code from auto-armor

### CODE REVIEW

**Comment by dominikaaaa** [comment]
We just follow Google style contentions loosely, if there's issues I'll autoformat and push it on your branch

**Comment by dominikaaaa** [comment]
I'll review this later, thanks

File: src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt [hunk 1]
```diff
@@ -25,137 +29,238 @@ class ElytraReplace : Module() {
     private val elytraFlightCheck = register(Settings.b("ElytraFlightCheck", true))
     private val logToChat = register(Settings.booleanBuilder("MissingWarning").withValue(false).build())
     private val playSound = register(Settings.booleanBuilder("PlaySound").withValue(false).withVisibility { logToChat.value }.build())
-    private val logThreshold = register(Settings.integerBuilder("MissingThreshold").withRange(1, 10).withValue(2).withVisibility { logToChat.value }.build())
+    private val logThreshold = register(Settings.integerBuilder("LogThreshold").withRange(1, 10).withValue(2).withVisibility { logToChat.value }.build())
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt:32]: why was this renamed?

Comment (Nathan-Webb) [reply src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt:32]: I felt that it would be a bit clearer to have the name in the GUI match the variable in the code. IMO `LogThreshold` conveys the idea of "the limit before elytra swaps start being logged in chat" better than `MissingThreshold`as elytras don't really go "missing".  `SwapThreshold` could work as well as reverting the change if its important to change names as little as possible.

Comment (dominikaaaa) [reply src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt:32]: It's supposed to be the amount of elytras you have left before it starts warning you in the chat. I'd just prefer a more intuitive name which isn't too long

Comment (Nathan-Webb) [reply src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt:32]: Do any of the following names work?
`ElytrasLeft`, `ElytraThreshold`, `ElytraLogLimit`.
If none of them work I can just revert it, no problems here!

Comment (dominikaaaa) [reply src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt:32]: I think `WarningThreshold` is most clear?

Comment (Nathan-Webb) [reply src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraReplace.kt:32]: Ah, that seems much clearer! I'll edit it asap 👍

**Comment by Nathan-Webb** [comment]
fuck, missclick

**Comment by dominikaaaa** [comment]
all G
